### PR TITLE
Add a template for MSAL UI Automation Pipeline

### PR DESCRIPTION
**Pull request by shahzaibj**
Some of the code to invoke run on test lab can be shared. This PR moves that common code to a template for sharing across multiple pipelines. Furthermore, it provides pipeline parameters for different devices that can be used for running UI Automation.

### CODE REVIEW

File: azure-pipelines/ui-automation/msal-broker-BrokerHost-nightly.yml [hunk 1]
```diff
@@ -147,54 +129,15 @@ stages:
...
-                  --test "$(Pipeline.Workspace)/msalautomationapks/$(msalAutomationAppTestApk)" \
-                  --device "model=${{ parameters.firebaseDeviceId }},version=${{ parameters.firebaseDeviceAndroidVersion }}" \
-                  --timeout "$(firebaseTimeout)" \
-                  --other-files \
-                  "/data/local/tmp/BrokerHost.apk=$(Pipeline.Workspace)/brokerHost/$(brokerHostApk),\
+      - template: ./templates/run-on-firebase.yml
+        parameters:
+          automationAppApkPath: "$(Pipeline.Workspace)/msalautomationapks/msalautomationapp-local-BrokerHost-debug.apk"
+          automationAppTestApkPath: "$(Pipeline.Workspace)/msalautomationapks/msalautomationapp-local-BrokerHost-debug-androidTest.apk"
+          testTargetPackages: ${{ parameters.testTargetPackages }}
```
Comment (fadidurah) [inline azure-pipelines/ui-automation/msal-broker-BrokerHost-nightly.yml:136]: Where are the test targets actually being set now that we removed --test-targets from the script?

**Comment by fadidurah** [comment]
Are the templates actively being fed into the active pipeline YML? So if you change something in the tempaltes, does it automatically apply to the pipeline files that are using that template? or is it just a template in the sense that someone can copy it and create a new pipeline yml file?

**Comment by shahzaibj** [comment]
> Are the templates actively being fed into the active pipeline YML? So if you change something in the tempaltes, does it automatically apply to the pipeline files that are using that template? or is it just a template in the sense that someone can copy it and create a new pipeline yml file?

Active. If you change something in the template then it affects all pipelines using that template.

File: azure-pipelines/ui-automation/msal-broker-BrokerHost-nightly.yml (hunk 1 above)
Comment (shahzaibj) [reply azure-pipelines/ui-automation/msal-broker-BrokerHost-nightly.yml:136]: They are still being supplied as pipeline parameters. Just coming from a higher level. The calling pipeline.

**Review by fadidurah — APPROVED** [review]

File: azure-pipelines/ui-automation/templates/build-msal-automation-app.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      - bash: |
+          echo "##vso[task.setvariable variable=ENV_VSTS_MVN_ANDROID_MSAL_ACCESSTOKEN]$(mvnAccessToken)"
+        displayName: Set MVN Access Token in Environment
+      - task: AzureKeyVault@2
+        displayName: 'Get Key vault AndroidAutomationRunnerAppSecret'
+        inputs:
+          azureSubscription: 'MSIDLABS_ANDROID_KV'
+          KeyVaultName: 'ADALTestInfo'
+          SecretsFilter: 'AndroidAutomationRunnerAppSecret'
+          RunAsPreJob: false
```
Comment (p3dr0rv) [inline azure-pipelines/ui-automation/templates/build-msal-automation-app.yml:43]: my recommendation will be to start removing all local mvnAccessToken and use $(AndroidAuthClient-Internal-Maven-Access-Token) from group variables AndroidAuthClientVariables, anyway this work can be realized in another PBI for all the pipelines.

**Review by p3dr0rv — APPROVED** [review]

File: azure-pipelines/ui-automation/templates/build-msal-automation-app.yml (hunk 2 above)
Comment (shahzaibj) [reply azure-pipelines/ui-automation/templates/build-msal-automation-app.yml:43]: Yes, can be explored as a separate effort for all pipelines.

**Pull request merged**